File: Capture the Flag/src/main/java/me/yoursole/ctf/Events/ItDamagePlayer.java
package me.yoursole.ctf.Events;

import me.yoursole.ctf.DataFiles.GameData;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;

public class ItDamagePlayer implements Listener {

    @EventHandler
    public void onPlayerDamage(EntityDamageByEntityEvent e){
        if(GameData.it!=null){
            if(e.getDamager().getType()== EntityType.PLAYER&& e.getDamager().getUniqueId()== GameData.it.getUniqueId()){
                double absorption = ((Player) e.getDamager()).getAbsorptionAmount();
                if(absorption<5){
                    ((Player) e.getDamager()).setAbsorptionAmount(((Player) e.getDamager()).getAbsorptionAmount()+0.5);
                }

            }
        }

    }
}

File: src/main/kotlin/me/yoursole/ctf/events/ItDamagePlayer.kt
package me.yoursole.ctf.events

import me.yoursole.ctf.datafiles.GameData
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent

object ItDamagePlayer : Listener {
    @EventHandler
    fun onPlayerDamage(e: EntityDamageByEntityEvent) {
        if (GameData.gameRunning) {
            if (e.damager is Player && e.damager.uniqueId === GameData.it!!.uniqueId) {
                val player = e.damager as Player
                if (player.absorptionAmount < 5) {
                    player.absorptionAmount += 0.5
                }
            }
        }
    }
}